src/components/ContentListTable/ContentListContext.tsx

```diff
@@ -0,0 +1,121 @@
+import { createContext, Dispatch, SetStateAction, useState } from 'react';
+import useDebounce from '../../services/useDebounce';
+import { useContentRepoParamsQuery } from '../../services/Content/ContentQueries';
+
+
+type ContentListContextProviderProps = {
+  children: React.ReactNode
+}
+
+interface RepoParams {
+  distribution_versions: Array<{ name: string, label: string }>
+  distribution_arches: Array<{ name:string, label: string }>
+}
+
+interface Labels {
+  versions: Array<string>,
+  arches: Array<string>
+}
+
+interface Context {
+  searchQuery: string,
+  debouncedSearchQuery: string,
+  setSearchQuery: Dispatch<SetStateAction<string>>,
+  filterType: string,
+  setFilterType: Dispatch<SetStateAction<string>>,
+  selectedVersions: Array<string>,
+  debouncedSelectedVersions: Array<string>,
+  setSelectedVersions: Dispatch<SetStateAction<Array<string>>>,
+  selectedArches: Array<string>,
+  debouncedSelectedArches: Array<string>
+  setSelectedArches: Dispatch<SetStateAction<Array<string>>>,
+  repoParams: RepoParams,
+  labels: Labels,
+}
+
+export const ContentListContext = createContext<Context>({
+  searchQuery: '',
+  debouncedSearchQuery: '',
+  setSearchQuery: () => null,
+  filterType: 'Name/URL',
+  setFilterType: () => null,
+  selectedVersions: [],
+  debouncedSelectedVersions: [],
+  setSelectedVersions: () => null,
+  selectedArches: [],
+  debouncedSelectedArches: [],
+  setSelectedArches: () => null,
+  repoParams: { distribution_versions: [], distribution_arches: [] },
+  labels: { versions: [], arches: [] },
+});
+
+const ContentListContextProvider = ({
+  children
+}: ContentListContextProviderProps) => {
+
+  const [filterType, setFilterType] = useState<string>('Name/URL');
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedVersions, setSelectedVersions] = useState<string[]>([]);
+  const [selectedArches, setSelectedArches] = useState<string[]>([]);
+
+  const filters = {
+    searchQuery, selectedVersions, selectedArches
+  }
+
+  const {
+    searchQuery: debouncedSearchQuery,
+    selectedVersions: debouncedSelectedVersions,
+    selectedArches: debouncedSelectedArches
+  } = useDebounce(filters)
+
+  const {
+    isLoading,
+    error,
+    isError,
+    data = { distribution_arches: [], distribution_versions: [] },
+  } = useContentRepoParamsQuery();
+
+  const getLabels = (type: string, names: Array<string>) => {
+    const namesLabels = type === 'arch' ? data['distribution_arches'] : data['distribution_versions'];
+    const labels: Array<string> = [];
+    names.forEach(name => {
+        const found = namesLabels.find(v => v.name === name )
+        if(found) {
+          labels.push(found.label)
+        }
+      }
+    );
+    return labels
+  };
+
+  const labels = {
+    versions: getLabels('version', debouncedSelectedVersions),
+    arches: getLabels('arch', debouncedSelectedArches)
+  }
+
+  if (isError) throw error
+
+  if (isLoading) return <></>;
```

Comment:
I have the loading logic built out on my branch, when this is merged I'll likely need to do something here.
```suggestion
  if (isLoading) return <></>; // TODO: Implement loader to match the top-level loader here.
```